trace-archiver-hdf5/src/main.rs

```diff
@@ -121,7 +125,25 @@ async fn main() -> Result<()> {
                                     .inc();
                             }
                         }
+                    } else if args.control_topic == Some(msg.topic().to_string()) {
+                        // A message has been received from the control topic.
+                        if run_start_buffer_has_identifier(payload) {
+                            debug!("New run start.");
+                            // Start recording trace data to file.
+                            if file.is_none() {
+                                let filename = generate_filename(msg.timestamp());
```

Comment:
The timestamp from the run start message should be used here, not the Kafka timestamp.